tests/common/widgets/current-refinements/routing.ts

```diff
@@ -0,0 +1,106 @@
+import { wait } from '@instantsearch/testutils';
+import {
+  createSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import { screen } from '@testing-library/dom';
+import type { CurrentRefinementsSetup } from '.';
+import { simple } from 'instantsearch.js/es/lib/stateMappings';
+import { history } from 'instantsearch.js/es/lib/routers';
+import type { Act } from '../../common';
+
+export function createRoutingTests(setup: CurrentRefinementsSetup, act: Act) {
+  describe('routing', () => {
+    describe('URLs created by widget', () => {
+      test('Consistently shows the right URL, even before widget is initialized', async () => {
```

Comment:
maybe a test with refinements too?

Reply:
You mean with `initialUiState` ? There is a `RefinementList` in this test which we interact with, but it tests what happens after the first render

Reply:
makes sense, in the breadcrumb test (not yet merged) I set an initial url